Makefile

```
@@ -7,7 +7,7 @@
 #
 
 .SUFFIXES:
-.SUFFIXES: .h .y .c .cpp .o
+.SUFFIXES: .hpp .cpp .y .o
```

I don't think any rules use `.hpp` or `.y` as suffixes?

```suggestion
.SUFFIXES: .cpp .o
```